File: docs-examples/example-java/src/test/java/io/pollend/quartz/docs/scheduler/MessageJob1.java
package io.pollend.quartz.docs.scheduler;

import org.quartz.Job;
import org.quartz.JobDataMap;
import org.quartz.JobExecutionContext;
import org.quartz.JobExecutionException;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class MessageJob1 implements Job {
    public List<String> messages = Collections.synchronizedList(new ArrayList<>());

    @Override
    public void execute(JobExecutionContext context) throws JobExecutionException {
        JobDataMap map = context.getJobDetail().getJobDataMap();
        messages.add(map.getString("message"));
    }
}

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/quartz/docs/scheduler/MessageJob1.kt
package io.micronaut.quartz.docs.scheduler

import org.quartz.Job
import org.quartz.JobExecutionContext
import org.quartz.JobExecutionException
import java.util.*

class MessageJob1 : Job {
    var messages = Collections.synchronizedList(ArrayList<String>())

    @Throws(JobExecutionException::class)
    override fun execute(context: JobExecutionContext) {
        val map = context.jobDetail.jobDataMap
        messages.add(map.getString("message"))
    }
}
